File: 20240807/Q1.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string main_str;
    string sub_str;

    cout << "Please input the string 1: ";
    getline(cin, main_str);
    cout << "Please input the string 2: ";
    getline(cin, sub_str);

    int count = 0;

    for (int i = 0; i <= main_str.length() - sub_str.length(); i++) {
        if (main_str[i] == sub_str[0]) { 
            bool match = true;
            for (int j = 0; j < sub_str.length(); j++) {
                if (main_str[i + j] != sub_str[j]) {
                    match = false;
                    break;
                }
            }
            if (match) {
                count++;
            }
        }
    }

    if (count > 0) {
        cout << "String 2 is a substring of string 1" << endl;
        cout << "Repeat " << count << " times" << endl;
    } else {
        cout << "String 2 is not a substring of string 1" << endl;
    }

    return 0;
}

File: 20240807/Q1.py
main_str = input("Please input the string 1:")
sub_str = input("Please input the string 2:")

count = 0

for i in range(len(main_str) - len(sub_str) + 1):
    if main_str[i:i+len(sub_str)] == sub_str:
        count += 1

if count > 0:
    print(f"String 2 is a substring of string 1")
    print(f"Repeat {count} times")
else:
    print(f"String 2 is not a substring of string 1")
